src/libraries/Common/src/System/Security/Cryptography/SLHDsa.cs

```diff
@@ -239,15 +150,88 @@ public bool VerifyData(ReadOnlySpan<byte> data, ReadOnlySpan<byte> signature, Re
                     SR.Argument_SignatureContextTooLong255);
             }
 
-            if (signature.Length != SignatureSizeInBytes)
+            if (signature.Length != Algorithm.SignatureSizeInBytes)
             {
                 return false;
             }
 
             return VerifyDataCore(data, context, signature);
         }
 
-        // TODO: VerifyPreHash
+        /// <summary>
+        ///   Sign the specified hash, writing the signature into the provided buffer.
+        /// </summary>
+        /// <param name="hash">
+        ///   The hash to sign.
+        /// </param>
+        /// <param name="destination">
+        ///   The buffer to receive the signature.
+        /// </param>
+        /// <param name="preHashAlgorithm"></param>
+        /// <param name="context">
+        ///   An optional context-specific value to limit the scope of the signature.
+        ///   The default value is an empty buffer.
+        /// </param>
+        /// <returns>
+        ///   The number of bytes written to the <paramref name="destination" /> buffer.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        ///   The buffer in <paramref name="destination"/> is too small to hold the signature.
+        /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="context"/> has a <see cref="ReadOnlySpan{T}.Length"/> in excess of
+        ///   255 bytes.
+        /// </exception>
+        /// <exception cref="ObjectDisposedException">
+        ///   This instance has been disposed.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   <para>The instance represents only a public key.</para>
+        ///   <para>-or-</para>
+        ///   <para>An error occurred while signing the hash.</para>
+        /// </exception>
+        public int SignPreHash(ReadOnlySpan<byte> hash, Span<byte> destination, HashAlgorithmName preHashAlgorithm, ReadOnlySpan<byte> context = default)
```

Comment:
Let's keep PreHash out for now.  While I have it in the issue, we need to see what the underlying providers are doing to know what shape we really want.

e.g. do we want HashAlgorithmName, Oid, both?